user/signer.go

```diff
@@ -8,65 +8,58 @@ import (
 }
 
-// SignerRFC6979 wraps [ecdsa.PrivateKey] and represents signer based on deterministic
-// ECDSA with SHA-256 hashing (RFC 6979). Provides [Signer] interface.
-//
-// Instances SHOULD be initialized with [NewSignerRFC6979] or [NewSignerRFC6979WithID].
-type SignerRFC6979 struct {
-	neofsecdsa.SignerRFC6979
-	userID ID
+type signer struct {
+	neofscrypto.Signer
+	usr ID
 }
 
-// NewSignerRFC6979 is a constructor for [SignerRFC6979].
-func NewSignerRFC6979(pk ecdsa.PrivateKey) *SignerRFC6979 {
-	var id ID
-	id.SetScriptHash((*keys.PublicKey)(&pk.PublicKey).GetScriptHash())
-
-	return &SignerRFC6979{
-		userID:        id,
-		SignerRFC6979: neofsecdsa.SignerRFC6979(pk),
-	}
+func (s signer) UserID() ID {
+	return s.usr
 }
 
-// NewSignerRFC6979WithID is a constructor for [SignerRFC6979] where you may specify [ID] associated with this signer.
-func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
-	return &SignerRFC6979{
-		SignerRFC6979: neofsecdsa.SignerRFC6979(pk),
-		userID:        id,
+// NewSigner combines provided [neofscrypto.Signer] and [ID] into [Signer].
+//
+// See also [NewAutoIDSigner].
+func NewSigner(s neofscrypto.Signer, usr ID) Signer {
+	return signer{
+		Signer: s,
+		usr:    usr,
 	}
 }
 
-// UserID returns the [ID] using script hash calculated for the given key.
-func (s SignerRFC6979) UserID() ID {
-	return s.userID
+func newAutoResolvedSigner(s neofscrypto.Signer, pubKey ecdsa.PublicKey) Signer {
+	var id ID
+	id.SetScriptHash((*keys.PublicKey)(&pubKey).GetScriptHash())
+
+	return NewSigner(s, id)
 }
 
-// StaticSigner emulates real sign and contains already precalculated hash.
-// Provides [Signer] interface.
-type StaticSigner struct {
-	neofscrypto.StaticSigner
-	id ID
+// NewAutoIDSigner returns [Signer] with neofscrypto.ECDSA_SHA512
+// signature scheme and user [ID] automatically resolved from the ECDSA public
+// key.
+//
+// See also [NewAutoIDSignerRFC6979].
+func NewAutoIDSigner(key ecdsa.PrivateKey) Signer {
+	return newAutoResolvedSigner(neofsecdsa.Signer(key), key.PublicKey)
 }
 
-// NewStaticSignerWithID creates new StaticSigner with specified [ID].
-func NewStaticSignerWithID(scheme neofscrypto.Scheme, sig []byte, pubKey neofscrypto.PublicKey, id ID) *StaticSigner {
-	return &StaticSigner{
-		StaticSigner: *neofscrypto.NewStaticSigner(scheme, sig, pubKey),
-		id:           id,
-	}
+// NewAutoIDSignerRFC6979 is an analogue of [NewAutoIDSigner] but with
+// [neofscrypto.ECDSA_DETERMINISTIC_SHA256] signature scheme.
+func NewAutoIDSignerRFC6979(key ecdsa.PrivateKey) Signer {
+	return newAutoResolvedSigner(neofsecdsa.SignerRFC6979(key), key.PublicKey)
 }
 
-// UserID returns underlying [ID].
-func (s *StaticSigner) UserID() ID {
-	return s.id
+// NewStaticSignerWithID creates new [Signer] with specified [ID].
+func NewStaticSignerWithID(scheme neofscrypto.Scheme, sig []byte, pubKey neofscrypto.PublicKey, id ID) Signer {
+	return NewSigner(neofscrypto.NewStaticSigner(scheme, sig, pubKey), id)
```

Comment:
@smallhive
having `NewSigner` and `neofscrypto.NewStaticSigner`: do we really need `NewStaticSignerWithID` then?

Reply:
We have both variants for the usual _signer_ in case if someone wants to set a different `user.ID`. I think we should save this flexibility here, as a minimum for now

Reply:
> save this flexibility here

to keep or not to keep `NewStaticSignerWithID`?

it may be
```go
s := user.NewStaticSignerWithID(scheme, sig, pub, id)
// or
s := user.NewSigner(neofscrypto.NewStaticSigner(scheme, sig, pub), id)
```

does first approach worth a separate function?

Reply:
The second option covers my requirements, you may remove _NewStaticSignerWithID_ or mark it _obsolete_

Reply:
ok, lets try to ride w/o it (we can always add if that)